linalg/solvers.cpp

```diff
@@ -1189,10 +1189,9 @@ void FGMRESSolver::Mult(const Vector &b, Vector &x) const
 
    final_norm = std::max(rel_tol*beta, abs_tol);
 
-   converged = false;
-
-   if (beta <= final_norm)
+   if (Monitor(0, beta, r, x) || beta <= final_norm)
    {
+      converged = false;
```

Comment:
It seems odd to set this one way only to set it the other in  a couple of lines.  Is there a side effect that I am missing here?

Reply:
This must be a merge oversight. It shouldn't be in the if statement body.

Reply:
Wrong fix, `converged = false` should be outside (otherwise it is uninitialized) and `converged = true` below.